### DISCUSSION TITLE
Why can we access LIST/DICT from outside of a function without passing it through Arguments | Python

### DISCUSSION THREAD

**Discussion by chienhsiang-hung** [community/community#34781] [Other Feature Feedback, Questions, & Ideas] [answered]
This question has been hanging on my mind quite a period. As title, does anyone know why we can access LIST/DICT from outside of a function without passing it through Arguments in Python? Thank you.

Foe example:
```python
test_list = []
def add_one():
  test_list.append(1)
  return
print(test_list)
```
`[1]`

**Comment by airtower-luna** [comment, marked as answer]
`test_list` is defined as a module-level ("global") variable. That makes it accessible everywhere within the module (the same source file). It's usually a bad idea to have modifiable state in module-level variables, but the language doesn't prevent it.

However, you shouldn't be seeing `[1]` as the output without *calling* `add_one()` first. I wonder if that's just missing in the example?

See also [Python Scopes and Namespaces](https://docs.python.org/3/tutorial/classes.html#tut-scopes) in the Python tutorial.

**Reply by chienhsiang-hung** [reply]
Thank you for kindly answering this!
And you're right. It should have been an `add_one()` before the `print()`.
